### ISSUE TITLE
Version 4.7.1221 creates no coverage compared to version 4.7.922

### ISSUE THREAD

**Issue by Lorilatschki** [OpenCover/opencover#1044]
Please provide the following information when submitting an issue. 
> Where appropriate replace the `[]` with a `[X]`

**NOTE** Support requests should initially be raised on [GitHub discussions](https://github.com/OpenCover/opencover/discussions) 

### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET 4.7
- [ ] .NET 4.7.1
- [ ] .NET 4.7.2
- [x] .NET 4.8
- [ ] .NET Core 2.2.0
- [ ] .NET Core 3.1.0
- [ ] .NET 5
- only release and LTS versions of .NET runtimes and SDKs will be supported 

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016
- [ ] Windows Server 2019

### I have already...

- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.
- [x] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [x] no coverage being recorded
- [x] 64 bit support

### Describe the bug
We try to switch from version `4.7.922` to `4.7.1221`. With the version 4.7.922 everything works as expected, line and branches are covered as expected. When I switch to the new version `4.7.1221` some assemblies getting zero coverage! The only thing which has changed between two runs is the OpenCover version, nothing else. 
We are using the following arguments: 
- `-register:path64` 
- `-filter:xxx`
- `-coverbytest:*.Test*.dll`
- `-returntargetcode:0`
- `-threshold:10`

Even when I remove all those arguments, not coverage is being created with version  `4.7.1221`!
```log
Tests run: 464, Errors: 0, Failures: 0, Inconclusive: 0, Time: 11,6616257 seconds
  Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0

Committing...
Visited Classes 0 of 4321 (0)
Visited Methods 0 of 36457 (0)
Visited Points 0 of 149799 (0)
Visited Branches 0 of 99897 (0)

==== Alternative Results (includes all methods including those without corresponding source) ====
Alternative Visited Classes 0 of 4459 (0)
Alternative Visited Methods 0 of 40190 (0)
```
We are running our tests with NUnit 2.6.2.

I have also tried version `4.7.1189` but it also behaves like `4.7.1221` and covers nothing.

### Steps to reproduce the problem:

- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf) again because we know you probably didn't do it the first time.
- **I am sorry, but I cannot share any code/assembly since it is confidential.**

**Comment by sjmla** [comment]
I ran into the same problem.

v4.7.922 is good, but v4.7.1189 is not ==> All my nunit tests are executed, but none visited.

In my case, I have 3 test DLLs, one of which gets coverage stats on both issues, while the other 2 do not on newer versions.

The difference between the two: the ones that stopped working load other mixed-mode assemblies that in turn load native DLLs.

**Comment by wowanator** [comment]
I can confirm this problem - I've discovered this only because the tool [Fine Code Coverage](https://github.com/FortuneN/FineCodeCoverage/) uses 4.922, while I tried to use the most recent one, which is 4.1221.
I'm experiencing this with the vstest console runner on .NET Framework 4.5.2.

**Comment by sawilde** [comment]
@Lorilatschki @sjmla @wowanator are you able to provide a sample project of it failing for you so I can replicate?

**Comment by Lorilatschki** [comment]
@sawilde sorry for the late response. Unfortunallty I am not able to provide such an sample project since the sources/assemblies are inner source. I can try to create one, but maybe @sjmla or @wowanator are able to share there projects in between.

**Comment by sjmla** [comment]
I tried to create a sample project, but, of course, cannot replicate the problem. 

So, either I am missing something, or I was wrong about mixed-mode assemblies having anything to do with it...

I'll attach the project just in case anyone else has better luck or want to pick up on this as a starting point...

[OpenCoverSample.zip](https://github.com/OpenCover/opencover/files/7042234/OpenCoverSample.zip)

**Comment by sjmla** [comment]
So, weirdly enough, after upgrading my old school project to sdk-style, and updating from VS2017 to VS2019, I can upgrade to the latest OpenCover and generate coverage reports without any issues.

@Lorilatschki, perhaps you can try that...?

**Comment by sawilde** [comment]
@sjmla that is a little freaky - can you provide more detail of the changes - perhaps the devil is in the detail.

**Comment by sjmla** [comment]
Not much in terms of details.

Ran
`dotnet tool install --global Project2015To2017.Migrate2019.Tool`, followed by
`dotnet migrate-2019 wizard my.sln`

Fixed breakages, and moved from using the VS2017-bundled MSBuild to use the VS2019 Build-Tools's MSBuild. That's it...

**Comment by Lorilatschki** [comment]
> So, weirdly enough, after upgrading my old school project to sdk-style, and updating from VS2017 to VS2019, I can upgrade to the latest OpenCover and generate coverage reports without any issues.
> 
> @Lorilatschki, perhaps you can try that...?

Sorry for the late response. But currently we are not able to migrate to VS2019 and the sdk style format. 
@sawilde is there a way I can debug this locally to identify the issue? Since you know you code best, can you point me to the code part's I need to a break points to in order to find out why the profiler does not catch any lines of code for some of our assemblies?

**Comment by sawilde** [comment]
> Since you know you code best,

I wish, most of this code was written 5+ years ago and I have not looked it at much in the past 3 years except for a few minor changes. However, I'll give it a go. 

@Lorilatschki I would start by using the debugview tool and see if anything useful pops out of there.

My suspicion is the PDB format is no longer supported by the library I am using to read them - the PDB format kept changing (or so it appeared) with each version of .net  and .net core - making managing OpenCover a bit of a 'mare. It's stabalised since which is why I suspect upgrading to the latest and greatest resolved the issue for others.

So I would look here (assuming you can build/debug OpenCover) - https://github.com/OpenCover/opencover/blob/master/main/OpenCover.Framework/Symbols/CecilSymbolManager.cs#L123

I don't have a windows machine anymore and so I am unable to support this project much more until I get a replacement.

**Comment by Lorilatschki** [comment]
@sawilde I had some time to analyse the issue. Here is what I currently found out:
**Regarding build**
>after installing all required software packages I was not able to build the solution out of the box. I had following two compile errors:

1. The [opencover.snk](https://github.com/OpenCover/opencover/blob/3f0fc6915863fb8c9f4d14e40967285a2a185d3d/main/OpenCover.Framework/OpenCover.Framework.csproj#L44) file does not exist in the [version](https://github.com/OpenCover/opencover/tree/3f0fc6915863fb8c9f4d14e40967285a2a185d3d/build/Version) folder! **Is it generated somehow?** _I solved this locally by duplicating the opencover.test.snk file and rename it to opencover.snk._
2. The profiler project also does not build from the scratch. It requires a[ version.h](https://github.com/OpenCover/opencover/blob/master/main/OpenCover.Profiler/OpenCover.Profiler.rc#L59) file which in fact also does not exist in the [version](https://github.com/OpenCover/opencover/tree/3f0fc6915863fb8c9f4d14e40967285a2a185d3d/build/Version) folder. _I solved this locally by removing the version.h include and setting the desired versions directly inside the [OpenCover.Profiler.rc](https://github.com/OpenCover/opencover/blob/master/main/OpenCover.Profiler/OpenCover.Profiler.rc#L62) file._

**Regarding the issue itself**
>I need some hints/advices, because it turned out, that **no exception** is thrown when covering a assembly and attaching a debugger to it. It looks like the assembly which is references by the test is not loaded by the memory profiler at all and thus nothing is collected. Therefor I added some console outputs to [MessageHandler.cs](https://github.com/OpenCover/opencover/blob/3f0fc6915863fb8c9f4d14e40967285a2a185d3d/main/OpenCover.Framework/Communication/MessageHandler.cs#L85). In the good case, the assemblies which are referenced by the test are loaded properly.
The only difference between the good and the bad case are, that in the bad case, the assembly to test refers a c++ dll. 
Is there a way to get a bit deeper inside the memory profiling mechanism, in order to find out why **no** StandardMessage with type MSG_Type.MSG_TrackAssembly for the bad case assembly is fired!?

**Comment by Lorilatschki** [comment]
> I believe that is manageable from the command line eg
> 
> `-communicationtimeout:<integer, e.g. 10000>`

Yes this is even better to use the command line cli instead of the env variable. After changing to a timeout of 3 minutes, it seems to work with the latest package.
Have you any idea why the profiler takes that long to instrument some of our assemblies in comparison to the version 4.7.922?

**Comment by sawilde** [comment]
Some work was done on excluding the hidden branches points that people get so upset about - I am surprised the impact would be that much though but then I don't have a .net project of large enough size to test it against anymore.
